### PR TITLE
refactor: claim flow tweaks

### PR DESCRIPTION
**Pull request by dohaki**
Minor UI tweaks + refactored hooks/mocks to resemble the final flow.

### CODE REVIEW

**Review by Tulun — COMMENTED** [review]
Overall fine, just a couple questions, but if this is supposed to be more in line with design QA Tim should look at it before final approval. (If this is still WIP, just lmk!)

File: src/views/Claim/hooks/useAirdropRecipient.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+async function getAirdropRecipient(account?: string) {
+  if (!account) {
+    return undefined;
+  }
+
+  const { data } = await axios.get<RecipientsWithProof>(
+    // TODO: replace with IPFS gateway url
+    "https://gist.githubusercontent.com/dohaki/53b241c26793260b6423fa1706e4cb96/raw/75bf9bb6e5adae3b059630de1b08b5db6d003959/recipients.json"
+  );
+
```
[the remark this answers is not in the document]
Comment (dohaki) [reply src/views/Claim/hooks/useAirdropRecipient.tsx:38]: Not sure yet. AFAIK it should either be IPFS or some other provider where we store this JSON.

File: src/views/Claim/hooks/useMerkleDistributor.tsx [hunk 2]
```diff
@@ -37,15 +53,25 @@ export function useMerkleDistributor() {
...
       // merkleDistributorContract
       //   .connect(provider)
       //   .hasClaimed(account)
-      //   .then(setHasClaimed);
-      mockedHasClaimed().then(setHasClaimed);
+      //   .then((hasClaimed) =>
+      //     setHasClaimedState({ status: "success", hasClaimed })
+      //   )
+      //   .catch((error) => setHasClaimedState({ status: "error", error }));
+      mockedHasClaimed()
```
[the remark this answers is not in the document]
Comment (dohaki) [reply src/views/Claim/hooks/useMerkleDistributor.tsx:70]: Yea exactly. I just adjusted it to mock both "already claimed" and "not claimed" states.

**Review by amateima — APPROVED** [review]
Looks good to me

**Pull request merged**